orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileManagerImpl.java

```diff
@@ -285,30 +285,76 @@ public OrcidProfile createOrcidProfile(OrcidProfile orcidProfile, boolean create
         return updatedTranslatedOrcid;
     }
 
-    private void addDefaultVisibilityToBioItems(OrcidProfile orcidProfile, Visibility defaultActivityVis) {
-        if (orcidProfile.getOrcidBio() != null){
-            if (orcidProfile.getOrcidBio().getExternalIdentifiers() != null)
-                for (ExternalIdentifier x : orcidProfile.getOrcidBio().getExternalIdentifiers().getExternalIdentifier() ){
-                    if (x.getVisibility() == null)
+    private void addDefaultVisibilityToBioItems(OrcidProfile orcidProfile, Visibility defaultActivityVis, Boolean isClaimed) {
+        if (defaultActivityVis == null) {
+            defaultActivityVis = Visibility.PRIVATE;
+        }
+
+        if(isClaimed == null) {
+            isClaimed = false;
+        }
+        
+        if (orcidProfile.getOrcidBio() != null) {
+            if (orcidProfile.getOrcidBio().getBiography() != null) {
+                if (isClaimed) {
+                    orcidProfile.getOrcidBio().getBiography().setVisibility(defaultActivityVis);
+                } else {
+                    Visibility visibility = orcidProfile.getOrcidBio().getBiography().getVisibility();
+                    orcidProfile.getOrcidBio().getBiography().setVisibility(visibility != null ? visibility : Visibility.PRIVATE);
+                }
+            }
+
+            if (orcidProfile.getOrcidBio().getExternalIdentifiers() != null) {
+                Visibility listVisibility = orcidProfile.getOrcidBio().getExternalIdentifiers().getVisibility();
+                for (ExternalIdentifier x : orcidProfile.getOrcidBio().getExternalIdentifiers().getExternalIdentifier()) {
+                    if (isClaimed) {
                         x.setVisibility(defaultActivityVis);
+                    } else {
+                        x.setVisibility(listVisibility != null ? listVisibility : Visibility.PRIVATE);
+                    }
                 }
-            if (orcidProfile.getOrcidBio().getKeywords() !=null)
-                for (Keyword x : orcidProfile.getOrcidBio().getKeywords().getKeyword() ){
-                    if (x.getVisibility() == null)
+            }
+            if (orcidProfile.getOrcidBio().getKeywords() != null) {
+                Visibility listVisibility = orcidProfile.getOrcidBio().getKeywords().getVisibility();
```

Comment:
What is list Vis, We are supposed to be getting activities Vis? Am I reading this wrong?